rclcpp/src/rclcpp/executor.cpp

```diff
@@ -679,6 +690,9 @@ void
 Executor::wait_for_work(std::chrono::nanoseconds timeout)
 {
   TRACEPOINT(rclcpp_executor_wait_for_work, timeout.count());
+  // To keep the guard conditions alive as they could be destroyed when the executor removes nodes.
+  // The lifetime of these guard conditions should be guaranteed during rcl_wait().
+  std::vector<GuardCondition::SharedPtr> alive_guard_condition_ptrs;
```

Comment:
@wjwwood 

Is it worth introducing this code not to break the CallbackGroup API and make it defer to create the guard condition in the callback group?
I was worried about adding extra code in this kind of the main loop. I appreciate it if you have any suggestions.

Reply:
@wjwwood friendly ping.